src/components/StatePicker.js

```diff
@@ -44,20 +38,27 @@ const defaultProps = {
     onBlur: () => {},
 };
 
-const StatePicker = forwardRef((props, ref) => (
-    <Picker
-        ref={ref}
-        inputID={props.inputID}
-        placeholder={{value: '', label: '-'}}
-        items={STATES}
-        onInputChange={props.onInputChange}
-        value={props.value}
-        label={props.label || props.translate('common.state')}
-        errorText={props.errorText}
-        onBlur={props.onBlur}
-        shouldSaveDraft={props.shouldSaveDraft}
-    />
-));
+const StatePicker = forwardRef((props, ref) => {
+    const STATES = _.map(props.translate('allStates'), ({stateISO, stateName}) => ({
+        value: stateISO,
+        label: stateName,
+    }));
+
+    return (
+        <Picker
+            ref={ref}
+            inputID={props.inputID}
+            placeholder={{value: '', label: '-'}}
+            items={STATES}
+            onInputChange={props.onInputChange}
+            value={props.value}
+            label={props.label || props.translate('common.state')}
```

Comment:
I am wondering, if the stored label is in one language, but we have switched languages, will this show in the wrong language? Though it seems we don't pass label at all?

Reply:
Ahh ok, got it, I thought the label was for the state itself